tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/dfd/ResourceProviderTest.java

```diff
@@ -0,0 +1,136 @@
+         
+         DFDURIResourceProvider dfduriResourceProvider = new DFDURIResourceProvider(dfdUri, ddUri);
+         dfduriResourceProvider.loadRequiredResources();
+         
+         this.dataDictionary = dfduriResourceProvider.getDataDictionary();
+         this.dataFlowDiagram = dfduriResourceProvider.getDataFlowDiagram();
+     }
+     
+	@Test
+	public void testCustomResourceProvider() {			        
+		DFDModelResourceProvider dfdModelResourceProvider = new DFDModelResourceProvider(this.dataDictionary, this.dataFlowDiagram);
+		
+		var analysis = new DFDDataFlowAnalysisBuilder().standalone()
+					.useCustomResourceProvider(dfdModelResourceProvider)
+					.usePluginActivator(Activator.class)
+					.modelProjectName(TEST_MODEL_PROJECT_NAME)
+	                .build();
+		
+		analysis.initializeAnalysis();
+		
+		var flowGraphs = analysis.findFlowGraphs();
+		flowGraphs.evaluate();
+		
+		assertEquals(flowGraphs.getTransposeFlowGraphs()
+                .size(), 4);
+	}
+	
+	@Test
+	public void testAbsolutePathForResourceProvider() {
+		String tempDir = System.getProperty("java.io.tmpdir");
+		var dfdFile = new File(tempDir,"BranchingTest.dataflowdiagram");
+		var ddFile = new File(tempDir,"BranchingTest.datadictionary");
+		dfdFile.deleteOnExit();
+		ddFile.deleteOnExit();
+		
+		ResourceSet resourceSet = new ResourceSetImpl();
+        Resource dfdResource = createAndAddResource(dfdFile.toString(), new String[] {"dataflowdiagram"}, resourceSet);
+        Resource ddResource = createAndAddResource(ddFile.toString(), new String[] {"datadictionary"}, resourceSet);
+
+        dfdResource.getContents()
+                .add(this.dataFlowDiagram);
+        ddResource.getContents()
+                .add(this.dataDictionary);
+
+        saveResource(dfdResource);
+        saveResource(ddResource);
+        
+        var analysis = new DFDDataFlowAnalysisBuilder().standalone()
+                .modelProjectName(TEST_MODEL_PROJECT_NAME)
+                .usePluginActivator(Activator.class)
+                .useDataFlowDiagram(dfdFile.toString())
+                .useDataDictionary(ddFile.toString())
+                .build();
+        
+        analysis.initializeAnalysis();
+        
+        var flowGraphs = analysis.findFlowGraphs();
+		flowGraphs.evaluate();
+		
+		assertEquals(flowGraphs.getTransposeFlowGraphs()
+                .size(), 4);
+	}
+	
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	private Resource createAndAddResource(String outputFile, String[] fileextensions, ResourceSet rs) {
+	     for (String fileext : fileextensions) {
+	        rs.getResourceFactoryRegistry().getExtensionToFactoryMap().put(fileext, new XMLResourceFactoryImpl());
+	     }		
+	     URI uri = URI.createFileURI(outputFile);
+	     Resource resource = rs.createResource(uri);
+	     ((ResourceImpl)resource).setIntrinsicIDToEObjectMap(new HashMap());
+	     return resource;
+	  }
+	
+	@SuppressWarnings({"unchecked", "rawtypes"})
```

Comment:
Could we get rid of this suppress warnings? It is probably caused by the usage of a class that has generics